src/AzureRMTools.ts

```diff
@@ -108,7 +108,14 @@ export class AzureRMTools {
 
     private getDeploymentTemplate(document: vscode.TextDocument): DeploymentTemplate | undefined {
         assert(document);
-        return this._deploymentTemplates.get(document.uri.toString());
+        const dt = this._deploymentTemplates.get(document.uri.toString());
+        if (dt && dt.documentVersion !== document.version) {
+            // tslint:disable-next-line: no-console
+            console.log("Document version does not match cached version");
+            return undefined;
```

Comment:
Returning undefined - is right thing to do when calling updateDeploymentTemplate?